JASP-Desktop/widgets/listmodel.cpp

```diff
@@ -77,38 +77,35 @@ void ListModel::_initTerms(const Terms &terms, const RowControlsOptions& allOpti
 	_rowControlsOptions = allOptionsMap;
 	endResetModel();
 
-	if (setupControlConnections && listView()->sourceModels().size() > 0)
-	{
-		QMap<ListModel*, Terms> map = getSourceTermsPerModel();
-		QMapIterator<ListModel*, Terms> it(map);
+	if (setupControlConnections)
+		for (QMLListView::SourceType* sourceType : listView()->sourceModels())
+			_connectSourceControls(sourceType->model, sourceType->usedControls);
+}
+
+
+
+void ListModel::_connectSourceControls(ListModel* sourceModel, const QSet<QString>& controls)
+{
+	// Connect option changes from controls in sourceModel that influence the terms of this model
+	if (!sourceModel || controls.size() == 0) return;
+
+	const Terms& terms = sourceModel->terms();
 
-		while (it.hasNext())
+	for (const QString& controlName : controls)
+	{
+		for (const Term& term : terms)
 		{
-			it.next();
-			ListModel* sourceModel = it.key();
-			const Terms& terms = it.value();
-			QMLListView::SourceType* sourceType = listView()->getSourceTypeFromModel(sourceModel);
-			if (sourceType)
+			JASPControlWrapper* control = sourceModel->getRowControl(term.asQString(), controlName);
+			BoundQMLItem* boundControl = dynamic_cast<BoundQMLItem*>(control);
+			if (boundControl && !_rowControlsConnected.contains(boundControl))
 			{
-				for (const QMLListView::SourceType::ConditionVariable& conditionVariable : sourceType->conditionVariables)
-					for (const Term& term : terms)
-						_connectSourceControl(sourceModel->getRowControl(term.asQString(), conditionVariable.controlName));
+				boundControl->boundTo()->changed.connect(boost::bind(&ListModel::_sourceTermsChangedHandler, this, _1));
```

Comment:
Why are you using boost connections instead of qt connections?

Reply:
Option class is not a Qt class.

Reply:
Ooh ja...